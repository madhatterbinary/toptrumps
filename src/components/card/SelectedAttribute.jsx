import React from 'react';
import PropTypes from 'prop-types';
import attributes from '../../constants/attributes';

const SelectedAttribute = (props) => {
  const { 
    selectedAttribute,
    attributeValue,
    isClosed,
    isWinner
  } = props;
  return selectedAttribute && isClosed ? (
    <div className="selected-attribute">
    {isWinner ?
      <span className="badge badge-success">{attributes[selectedAttribute]}:
    {attributeValue}</span> :
     <span>    {attributes[selectedAttribute]}:
    {attributeValue}</span>
    }
    </div>
  ) : (
    <div className="selected-attribute">Select an attribute</div>
  );
}

SelectedAttribute.propTypes = {
  selectedAttribute: PropTypes.string,
  attributeValue: PropTypes.string,
}

export default SelectedAttribute;
