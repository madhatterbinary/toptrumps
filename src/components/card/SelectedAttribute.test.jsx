import React from 'react';
import SelectedAttribute from './SelectedAttribute';
import TestRenderer from 'react-test-renderer';

describe('SelectedAttribute', () => {
  it('renders', () => {
    const component = TestRenderer.create(<SelectedAttribute selectedAttribute="" attributeValue="" status="" isClosed={true} isWinner={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
