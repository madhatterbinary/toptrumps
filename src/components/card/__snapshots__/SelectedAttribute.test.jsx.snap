// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedAttribute renders 1`] = `
<div
  className="selected-attribute"
>
  Select an attribute
</div>
`;
